drivers/eeprom/eeprom_st25dv.c

```diff
@@ -43,7 +43,86 @@ LOG_MODULE_REGISTER(eeprom_st25dv);
 #define ST25DV_REG_IC_REF      0x0017
 #define ST25DV_REG_UUID        0x0018
 #define ST25DV_REG_IC_REV      0x0020
-#define ST25DV_REG_I2C_PWD     0x0900
+#define ST25DV_REG_I2C_PASSWD  0x0900
+#define ST25DV_REG_I2C_SSO_DYN 0x2004
+
+/** @brief Length of the MEM_SIZE field of the ST25DV */
+#define ST25DV_MEM_SIZE_LENGTH   (2u)
+/** @brief Length of the BLK_SIZE field of the ST25DV */
+#define ST25DV_BLK_SIZE_LENGTH   (1u)
+/** @brief Length of the IC_REV field of the ST25DV */
+#define ST25DV_IC_REV_LENGTH     (1u)
+/** @brief Length of ENDA registers */
+#define ST25DV_ENDA_LENGTH       (1u)
+/** @brief Length of I2C_PASSWD message */
+#define ST25DV_I2C_PASSWD_LENGTH (17u)
+/** @brief Length of RFAxSS registers */
+#define ST25DV_RFASS_LENGTH      (1u)
+
+/**
+ * @brief I2C Time out (ms), max value : tw + tw * (Max write bytes) / (Internal page write) :  5 +
+ * 5*(256/16).
+ */
+#define EEPROM_ST25DV_WRITE_TIMEOUT 85U
+
+/* I2C_SSO_Dyn */
+#define EEPROM_ST25DV_I2C_SSO_DYN_I2CSSO_SHIFT (uint8_t)(0)
```

Comment:
Why (uint8_t)?

Reply:
(uint8_t) removed

Reply:
Still, why did you want to cast it? if there is a valid reason, this should be written differently. In most cases, 0U or 0UL would be unambiguous.

Reply:
Some of those fields were copied and modified from ST's Linux driver. I tend to use () myself so I left it that way but the cast did not make sense I think.

Reply:
> Some of those fields were copied and modified from ST's Linux driver. I tend to use () myself so I left it that way but the cast did not make sense I think.

Please be aware that copying from e.g. GPL licensed code to an Apache-2.0 licensed file is not permitted by the GPL.

Reply:
I have rewritten the copied code and comments. It should be fine now.